### PR TITLE
Move the age field to an unsigned block

### PR DESCRIPTION
**Pull request by anoadragon453**
The spec is out of date on this one. Bug filed [here](https://github.com/matrix-org/matrix-doc/issues/1294).

App Services expect this behavior.

### CODE REVIEW

File: appservice.go [hunk 1]
```diff
@@ -15,10 +15,16 @@
 
 package gomatrixserverlib
 
+// ApplicationServiceUnsigned is the contents of the unsigned field of an
+// ApplicationServiceEvent
+type ApplicationServiceUnsigned struct {
```
[the remark this answers is not in the document]
Comment (anoadragon453) [reply appservice.go]: Despite what the spec might say, an event to an application service actually looks like the following:

```
    {
      "origin_server_ts": 1528988734374,
      "sender": "@testing-2bwv:localhost",
      "event_id": "$OMpf52ThzC69stAe:localhost",
      "content": {
        "membership": "join",
        "displayname": "bob"
      },
      "room_id": "!Thyj9JUx6phZqTQW:localhost",
      "unsigned": {
        "age": 33,
      },
      "user_id": "@testing-2bwv:localhost",
      "type": "m.room.member"
    }
```

We specify the `age` field as unsigned as it is constantly changing and we don't want to resign it every time we need to resend a dropped event. Application services, like the discord bridge, expect the event like this.

[the remark this answers is not in the document]
Comment (erikjohnston) [reply appservice.go]: I meant that particular struct? Defining it without using it in the code feels weird

[the remark this answers is not in the document]
Comment (anoadragon453) [reply appservice.go]: Ah, I see yes. It *does* get used as RawJSON, but it would be better to assert that that struct is what's used.

**Comment by anoadragon453** [comment]
Made the change, will need to update https://github.com/matrix-org/dendrite/pull/477 once this is merged.

**Pull request merged**